packages/runtime/container-runtime/src/blobManager/blobManager.ts

```diff
@@ -374,8 +379,20 @@ export class BlobManager {
 			storageId = blobId;
 		} else {
 			const attachedStorageId = this.redirectTable.get(blobId);
-			assert(!!attachedStorageId, 0x11f /* "requesting unknown blobs" */);
```

Comment:
well probably need something more robust here which can differentiate between deferred vs non-deferred handles

Reply:
Agreed it will be needed long-term (particularly thinking about observability), but are you saying you think it's also needed for the more immediate problem we're trying to solve?  Is your main concern around keeping this assert enabled for non-deferred handles?

Reply:
yeah. i think we should be able to differentiate between the new vs old handles

Reply:
OK but do you have a particular purpose in mind?  I think you are hinting that you have a goal in mind for this capability, want to make sure I'm on the same page

Reply:
the immediate goal is to keep the assert, or an error for blobs which we know should exists, and only suppress when there are blobs with the new behavior

Reply:
This was resolved by #24224